### ISSUE TITLE
iOS app is very slow to react to network state change

### ISSUE THREAD

**Issue by mhsmith** [Electron-Cash/Electron-Cash#971]
When I put my iPhone into airplane mode, Electron Cash continued displaying "Online" for about 5 minutes.

When I turned off airplane mode, the phone reconnected to WiFi immediately, but Electron Cash kept displaying "Offline" for about 30 seconds.

The phone has no SIM card, so the mobile network isn't a factor.

**Comment by cculianu** [comment]
Ah ok. I noticed that when developing and never went back to figure out how to handle it better. 

 I basically just rely on whatever the existing network layer says/does -- and at least on my machine it performs equally as bad even on desktop Qt.  

See attachment.  I left the network off for a full minute and it's still green (macOS wifi icon is in the top left in this pic, set to off, yet EC's network status still says online).

<img width="409" alt="screen shot 2018-11-22 at 2 39 58 am" src="https://user-images.githubusercontent.com/266627/48875179-f05efb00-edff-11e8-9fa5-7337c1e53f62.png">

So -- all I'm saying is -- I didn't do anything particularly incompetent ;).  

But perhaps on iOS there are things I can do to speed up the detection of the network going away.

Thanks for reporting this and will keep it on the table as a UX thing that should definitely be fixed.

**Comment by mhsmith** [comment]
Yeah, I thought it probably affected all platforms: I'll rename the issue.

**Comment by cculianu** [comment]
Well it's a particularly painful issue on mobile since you're more likely to go in and out of network service, airplane mode, etc.. but yeah it's a general "all platform" issue.

**Comment by mhsmith** [comment]
@cculianu: I see you've reduced the re-connection timeouts for iOS, so I've done the same for Android.

However, it still often doesn't detect a disconnection until the next ping message, so I've reduced the ping interval on Android from 5 minutes to 1. I don't know whether any other platforms could benefit from this, so I'll leave the issue open.

**Comment by cculianu** [comment]
Yeah I forget why I reduced it on iOS -- likely just to keep things alive.. but yeah sounds good.
